composer/callbacks/memory_monitor.py

```diff
@@ -133,7 +133,8 @@ def init(self, state: State, logger: Logger) -> None:
         # Not relying on `torch.cuda.is_available()` since the model could be on CPU.
         model_device = next(state.model.parameters()).device
 
-        if model_device.type not in ('cuda', 'meta'):
+        print('----UNGA BUNGA', model_device.type)
+        if model_device.type == 'cpu':
```

Comment:
Can you try moving the original warning to after_load? At that point, I think the model should be on GPU only if using mixed init